src/middleware/normalize.rs

```diff
@@ -10,20 +10,29 @@ use regex::Regex;
 use crate::service::{ServiceRequest, ServiceResponse};
 use crate::Error;
 
-#[derive(Default, Clone, Copy)]
+/// To be used when constructing `NormalizePath` to define it's behavior.
+#[derive(Clone, Copy)]
+pub enum TrailingSlash {
```

Comment:
Can you stick a #[non_exhaustive] on this please. Unlikely folks will want to match on it but would let us do things in future without breaking changes.